config/crds/vsphereproviderconfig_v1alpha1_vsphereclusterproviderconfig.yaml

```diff
@@ -42,5 +36,5 @@ status:
   acceptedNames:
     kind: ""
     plural: ""
-  conditions: []
-  storedVersions: []
+  conditions: null
```

Comment:
IIRC the `null` here causes issues.
This generated CRD has this diff but was intentionally not merged because of the `null` issue.
I always thought this was because of some incorrect version of the tooling on my local machine.
Can you verify once that this change will not cause an issue when deploying the CAPV?

Reply:
Hi @sidharthsurana,

I won't get into the details, but we spent some time (@frapposelli, @figo, and myself) figuring out the last time someone actually generated the CRDs ahead of committing them. It wasn't since the first commit. They're very stale. We used `kustomize` v2 exactly, and we all generated the same deltas. If this causes a failure, then the failure is in the project or its settings, because there's currently a bug in CI where the CRDs aren't actually generated properly on the server-side due to the CI jobs not including or downloading `kustomize`.

If the generated CRD is invalid, then the sources are invalid or the tooling is invalid. Either way, if there is *any* **manual** step between generating the CRD and using the CRD, then that's a bug.

Reply:
> Can you please verify once that this change will not cause an issue when deploying the CAPV?

Hi @sidharthsurana,

If it does, then good, because if generating the CRDs causes CAPV to fail, then then either the CRD sources, the tooling, or the process needs to be fixed. It has to be manual or generated -- it can't be both.

Reply:
I agree, but the point is, if this CRD breaks things, then we should fix the source code so that it generates the right CRD rather than checking in CRD that may not work.